src/mdoc/Verifier.ts

```diff
@@ -299,76 +313,156 @@ export class Verifier {
+   * @returns {Verifier}
+   */
+  public usingSessionTranscriptForOID4VP(
+    mdocGeneratedNonce: string,
+    clientId: string,
+    responseUri: string,
+    verifierGeneratedNonce: string,
+  ): Verifier {
+    this.usingSessionTranscriptBytes(
+      cborEncode(
+        DataItem.fromData([
+          null, // deviceEngagementBytes
+          null, // eReaderKeyBytes
+          [
+            createHash('sha256').update(cborEncode([clientId, mdocGeneratedNonce])).digest(),
+            createHash('sha256').update(cborEncode([responseUri, mdocGeneratedNonce])).digest(),
+            verifierGeneratedNonce,
+          ],
+        ]),
+      ),
+    );
+    return this;
+  }
+
+  /**
+   * Set the session transcript data to use for the verification as defined in ISO/IEC 18013-7 in Annex A (Web API), 2024 draft.
+   *
+   * This should match the session transcript as it will be calculated by the mdoc app.
+   *
+   * @param {Buffer} deviceEngagementBytes - The device engagement, encoded as a Tagged 24 cbor
+   * @param {Buffer} readerEngagementBytes - The reader engagement, encoded as a Tagged 24 cbor
+   * @param {Buffer} eReaderKeyBytes - The reader ephemeral public key as a COSE Key, encoded as a Tagged 24 cbor
+   * @returns {Verifier}
+   */
+  public usingSessionTranscriptForWebAPI(
+    deviceEngagementBytes: Buffer,
+    readerEngagementBytes: Buffer,
+    eReaderKeyBytes: Buffer,
+  ): Verifier {
+    this.usingSessionTranscriptBytes(
+      cborEncode(
+        DataItem.fromData([
+          new DataItem({ buffer: deviceEngagementBytes }),
+          new DataItem({ buffer: eReaderKeyBytes }),
+          createHash('sha256').update(readerEngagementBytes).digest(),
+        ]),
+      ),
+    );
+    return this;
+  }
+
+  /**
+   * @param {Buffer} ephemeralReaderKey - The private part of the ephemeral key used in the session where the DeviceResponse was obtained. This is only required if the DeviceResponse is using the MAC method for device authentication.
+   * @returns {Verifier}
+   */
+  public usingEphemeralReaderKey(ephemeralReaderKey: Uint8Array): Verifier {
+    this.#ephemeralReaderKey = ephemeralReaderKey;
+    return this;
+  }
+
+  /**
+   * Disables the certificate validation
+   * @returns {Verifier}
+   */
+  public disableCertificateChainValidation(): Verifier {
+    this.#disableCertificateChainValidation = true;
+    return this;
+  }
+
   /**
    * Parse and validate a DeviceResponse as specified in ISO/IEC 18013-5 (Device Retrieval section).
    *
    * @param encodedDeviceResponse
-   * @param options.encodedSessionTranscript The CBOR encoded SessionTranscript.
-   * @param options.ephemeralReaderKey The private part of the ephemeral key used in the session where the DeviceResponse was obtained. This is only required if the DeviceResponse is using the MAC method for device authentication.
+   * @param options.disableCertificateChainValidation Whether to use the certificate validation.
```

Comment:
Is this expected? It seems like you are removing the options param

Reply:
👍🏼 Removed in 10fe386